source/common/upstream/load_balancer_impl.h

```diff
@@ -165,6 +149,11 @@ class LoadBalancerBase : public LoadBalancer {
 
 class LoadBalancerContextBase : public LoadBalancerContext {
 public:
+  static bool validateOverrideHostStatus(Host::Health health, OverrideHostStatus status);
+
+  static HostConstSharedPtr selectOverrideHost(const HostMap* host_map,
+                                               LoadBalancerContext* context);
```

Comment:
I wonder if it makes sense to have this be on the LoadBalancerBase, it just seems a tad bit odd to have a static function on a class that accepts a pointer to said class. Not a big deal if you prefer it here

Reply:
I haven't found out a better position to place these util functions. I have no particular insistence on this, I think we can move it as needed in the future.